diff --git a/opencompass/datasets/livecodebench/evaluator.py b/opencompass/datasets/livecodebench/evaluator.py
index 9ae936a..62c3da0 100644
--- a/opencompass/datasets/livecodebench/evaluator.py
+++ b/opencompass/datasets/livecodebench/evaluator.py
@@ -6,6 +6,7 @@ from concurrent.futures import ProcessPoolExecutor, as_completed
 
 import numpy as np
 from tqdm import tqdm
+from typing import List
 
 from opencompass.openicl.icl_evaluator import BaseEvaluator
 from opencompass.registry import ICL_EVALUATORS
@@ -115,7 +116,7 @@ def evaluate_generations_by_problem(problem_generations: list, sample: list,
 
 def evaluate_generations(
     samples_list: list,
-    generations_list: list[list[str]],
+    generations_list: List[List[str]],
     debug: bool = False,
     num_process_evaluate: int = 16,
     timeout=6,
@@ -266,7 +267,7 @@ class LCBCodeGenerationEvaluator(BaseEvaluator):
         return metrics
 
 
-def evaluate_score(args) -> list[bool]:
+def evaluate_score(args) -> List[bool]:
     gs, (c, i, o) = args
 
     execution_results = []
diff --git a/opencompass/models/openai_api.py b/opencompass/models/openai_api.py
index ab7ab30..f71e313 100644
--- a/opencompass/models/openai_api.py
+++ b/opencompass/models/openai_api.py
@@ -473,17 +473,17 @@ class OpenAISDK(OpenAI):
         query_per_second: int = 1,
         rpm_verbose: bool = False,
         retry: int = 2,
-        key: str | List[str] = 'ENV',
-        org: str | List[str] | None = None,
-        meta_template: Dict | None = None,
+        key: Union[str, List[str]] = 'ENV',
+        org: Union[str, List[str], None] = None,
+        meta_template: Union[Dict, None] = None,
         openai_api_base: str = OPENAI_API_BASE,
         openai_proxy_url: Optional[str] = None,
         mode: str = 'none',
-        logprobs: bool | None = False,
-        top_logprobs: int | None = None,
-        temperature: float | None = None,
-        tokenizer_path: str | None = None,
-        extra_body: Dict | None = None,
+        logprobs: Union[bool, None] = False,
+        top_logprobs: Union[int, None] = None,
+        temperature: Union[float, None] = None,
+        tokenizer_path: Union[str, None] = None,
+        extra_body: Union[Dict, None] = None,
         max_completion_tokens: int = 16384,
         verbose: bool = False,
         status_code_mappings: dict = {},
@@ -524,7 +524,7 @@ class OpenAISDK(OpenAI):
             self.logger.info(f'Used openai_client: {self.openai_client}')
         self.status_code_mappings = status_code_mappings
 
-    def _generate(self, input: PromptList | str, max_out_len: int,
+    def _generate(self, input: Union[PromptList, str], max_out_len: int,
                   temperature: float) -> str:
         from openai import BadRequestError
         assert isinstance(input, (str, PromptList))
diff --git a/requirements/runtime.txt b/requirements/runtime.txt
index cfbde9c..dc42193 100644
--- a/requirements/runtime.txt
+++ b/requirements/runtime.txt
@@ -1,45 +1,10 @@
-absl-py
-accelerate>=0.19.0
-cpm_kernels
-datasets>=2.12.0
-einops==0.5.0
-evaluate>=0.3.0
-func_timeout
-fuzzywuzzy
-h5py
-huggingface_hub<=0.24.7
-immutabledict
-importlib-metadata
-jieba
-json5
-mmengine-lite
 nltk==3.8
-numpy>=1.23.4,<2.0.0
-openai
-OpenCC
-opencv-python-headless
-pandas<2.0.0
+importlib_metadata
 prettytable
-protobuf
-pyext
-python-Levenshtein
+jieba
+rouge_chinese
 rank_bm25==0.2.2
-rapidfuzz
-requests>=2.31.0
-retrying
-rich
-rouge
--e git+https://github.com/Isaac-JL-Chen/rouge_chinese.git@master#egg=rouge_chinese
-rouge_score
-sacrebleu
-scikit_learn==1.5.0
-seaborn
 sentence_transformers==2.2.2
-tabulate
-tiktoken
-timeout_decorator
-tokenizers>=0.13.3
-torch>=1.13.1
-tqdm>=4.64.1
-transformers>=4.29.1
-typer
+fuzzywuzzy
+h5py
+huggingface_hub == 0.25.0
\ No newline at end of file
